Mono repo changes

jonathanmeier5 opened the pull request:
This is a sizable PR. It contains the following changes:

- support for overriding a custom config base name:
interface: https://github.com/openslate/compose-flow/pull/79/files#diff-4c6886de6fb0f1494eb05192ceab17d1R124

- support for interpolating `compose-flow.yml` and generating a new `compose-flow-<env>-<config-name>-config.yml` file that is used in place of the old, un-interpolated `compose-flow.yml`
core functionality: https://github.com/openslate/compose-flow/pull/79/files#diff-74e29ff7927b3648d3c7fc0a0bacaecbR93

- support for layering environments on top of each other
documented: https://github.com/openslate/compose-flow/pull/79/files#diff-0516e8fbe588e7299adbb064a6f40715R21
functionality: https://github.com/openslate/compose-flow/pull/79/files#diff-0516e8fbe588e7299adbb064a6f40715R400

marcusianlevine commented on src/compose_flow/commands/workflow.py:
```diff
@@ -36,11 +37,16 @@
 class Workflow(object):
     def __init__(self, argv=None):
-        self.argv = argv or sys.argv[1:]
+        self.argv = argv
```
Was this change just for clarity?

If so, I think the most Pythonic way to do this would be with a ternary: `self.argv = argv if argv is not None else sys.argv[1:]`

jonathanmeier5 replied:
Agreed 👍

marcusianlevine replied:
So wait did you change this? This thread was resolved but the code is still the same...

marcusianlevine commented on src/compose_flow/commands/workflow.py:
```diff
@@ -213,12 +235,30 @@ def _set_arg_defaults(self):
+                if config_name.startswith(prefix):
+                    self.config_basename = config_name.split(prefix, 1)[-1]
+        else:
+            # both config_name and config_basename are not set
```
Shouldn't this raise an error? Not sure how this is supposed to work when `self.config_basename == None`

jonathanmeier5 replied:
I don't think so. Previously we allowed `self.args.config_name` to be `None`. If it was `None`, then we executed this line [here](https://github.com/openslate/compose-flow/pull/79/files/772a91629552d88af752bf663d589521dacf1c63#diff-4c6886de6fb0f1494eb05192ceab17d1L221). We execute [these lines](https://github.com/openslate/compose-flow/pull/79/files/772a91629552d88af752bf663d589521dacf1c63#diff-4c6886de6fb0f1494eb05192ceab17d1L218) over [here](https://github.com/openslate/compose-flow/pull/79/files/772a91629552d88af752bf663d589521dacf1c63#diff-4c6886de6fb0f1494eb05192ceab17d1R239) now regardless.

Unless I'm missing something?

marcusianlevine commented on src/compose_flow/kube/mixins.py:
```diff
@@ -40,11 +56,25 @@ class KubeMixIn(object):
+    @property
+    @lru_cache()
+    def rendered_config(self):
+        config = self.config
```
This assignment feels unnecessary, just pass `self.config` like you pass `self.workflow`

jonathanmeier5 commented:
@marcusianlevine touched up and addressed your comments.

See test plan/execution here: https://openslate.myjetbrains.com/youtrack/issue/OS-2340

data-jobs and fb-fe built fine. Deployments worked fine as well.

I've noted a few known bugs in the testing ticket, but I'd rather not let them block on this whole PR.
